### ISSUE TITLE
Migrate from CircularJSON to Flatted

### ISSUE THREAD

**Issue by alexeyinkin** [akvelon/dart-highlighting#44]
https://github.com/WebReflection/flatted

**Comment by yescorp** [comment]
As I found out, Flatted probably doesn't do anything for us. After we parse the stringified object with Flatted we get the initial circular object.

Its scope of use is probably parsing circular JSON objects between services.

**Comment by alexeyinkin** [comment]
Yes. We actually do not need the perfect serialize-deserialize pass. We serialize with CircularJSON but deserialize with the ordinary JSON and then look for CircularJSON-specific tokens. This works because CircularJSON produces valid JSON syntax.

They deprecated CircularJSON in favor of Flatted, but it uses its custom syntax and so cannot be deserialized with JSON.parse(). So it will not fit the workflow.
